### DISCUSSION TITLE
Issue with SD Card Logging Task Failing to Open File When UWB Deck is Active

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1613] [Q&A]
**Setup and Hardware**:
- **Crazyflie 2.1** with **SD Card Deck** and **UWB Deck** attached.
- Both decks share the same SPI bus.
- The SD card logging task has been configured to write data at 50 Hz.

**Problem**:
When the UWB deck is plugged in, I encounter an issue where the SD card logging task fails to open files for logging. Without the UWB deck, logging works as expected. I suspect that concurrent access to the SPI bus by both the SD and UWB tasks is causing conflicts.

**Technical Details**:
- **Task Priorities**:
   - UWB Task: Priority 3
   - SD Card Logging Task: Priority 0
- **DMA is enabled** for SPI transactions in both tasks.

**Questions**:
- Has anyone else encountered similar issues with concurrent SPI access in Crazyflie when using both UWB and SD Card decks?
- Are there recommended patterns or configurations to manage shared SPI bus access between tasks of different priorities, especially when both use DMA?
- Would implementing a shared mutex for SPI transactions be enough, or are there other recommended strategies for ensuring reliable SD card access?

Any insights or suggestions would be greatly appreciated!

**Comment by tobbeanton** [comment]
This has been working before but maybe something has caused it to break. What mode are you running the Loco deck in? TWR is very chatty, which doesn't help the situation. Is using the alternative SPI bus on the SD-card an option?

**Reply by amitx64** [reply]
@tobbeanton I am using the Loco deck in TDOA3 mode. It works fine after I disable the UWB task before writing.

**Reply by tobbeanton** [reply]
Could you try the [alternative SPI bus](https://www.bitcraze.io/products/micro-sd-card-deck/#development)?

**Reply by amitx64** [reply]
@tobbeanton ok I will try.
